persistence/src/commonMain/kotlin/com/wire/kalium/persistence/backup/DatabaseExporter.kt

```diff
@@ -48,83 +47,94 @@ interface DatabaseExporter {
 internal class DatabaseExporterImpl internal constructor(
     user: UserIDEntity,
     private val platformDatabaseData: PlatformDatabaseData,
-    private val dumpContentQueries: DumpContentQueries,
-    private val sqlDriver: SqlDriver,
-    private val isDataEncrypted: Boolean,
+    private val localDatabase: UserDatabaseBuilder
 ) : DatabaseExporter {
 
     private val backupUserId = user.copy(value = "backup-${user.value}")
 
     @Suppress("TooGenericExceptionCaught", "ReturnCount")
-    override fun exportToPlainDB(): String? {
+    override fun exportToPlainDB(localDBPassphrase: UserDBSecret?): String? {
         // delete the backup DB file if it exists
         if (deleteBackupDBFile()) {
             return null
         }
 
-        // create a new backup DB file
+        // create a new backup DB file with empty passphrase
+        // this will force the app to use sql Cipher and enable the encrypted local DB
+        // to be attached to the plain DB
+        // also unencrypted DB (in debug mode) can be attached to the plain DB
+        // win win
         val plainDatabase: UserDatabaseBuilder =
-            userDatabaseBuilder(platformDatabaseData, backupUserId, null, KaliumDispatcherImpl.io, false)
-        plainDatabase.sqlDriver.close()
-
-        val plainDBPath = plainDatabase.dbFileLocation() ?: run {
-            kaliumLogger.e("Failed to get the plain DB path")
-            return null
+            userDatabaseBuilder(
+                platformDatabaseData,
+                backupUserId,
+                UserDBSecret(ByteArray(0)),
+                KaliumDispatcherImpl.io,
+                false
+            )
+
+        // check the plain DB path and return null if it was not created successfully
+        plainDatabase.dbFileLocation().also {
+            if (it == null) {
+                kaliumLogger.e("Failed to get the plain DB path")
+                return null
+            }
         }
 
         // copy the data from the user DB to the backup DB
+        if (!attachLocalToPlain(localDatabase, plainDatabase, localDBPassphrase)) {
+            plainDatabase.sqlDriver.close()
+            deleteBackupDBFile()
+            return null
+        }
 
         try {
-            sqlDriver.execute(null, "BEGIN", 0)
-            attachDatabase(isDataEncrypted, plainDBPath)
-            dumpContent()
-            sqlDriver.execute(null, "COMMIT", 0)
+            // attach the plain DB to the user DB
```

Comment:
The attachment is actually done in the `if` above right?

Reply:
yes